### ISSUE TITLE
Error thrown when opening channel with Frame

### ISSUE THREAD

**Issue by kalinon** [cloudamqp/amqp-client.cr#4]
It seems that there is a bug when trying to open a channel that passes `AMQ::Protocol::Frame`. Errors are not created with other types (`String`, `Int32`, etc) but this simple test shows the error:

`test.cr`
```
require "amqp-client"
channel = Channel(AMQ::Protocol::Frame).new
channel.receive
```

Output:
```
crystal run test.cr
Showing last frame. Use --error-trace for full trace.

In /usr/local/Cellar/crystal/0.31.0/src/channel.cr:167:20

 167 | @receivers << {Fiber.current, pointerof(value), pointerof(state), nil}
                  ^-
Error: no overload matches 'Deque(Tuple(Fiber, Pointer(AMQ::Protocol::Frame), Pointer(Channel::DeliveryState), Channel::SelectContext(AMQ::Protocol::Frame) | Nil))#<<' with type Tuple(Fiber, Pointer(AMQ::Protocol::Frame), Pointer(Channel::DeliveryState), Nil)

Overloads are:
 - Deque(T)#<<(value : T)
```

Crystal Env:
```
CRYSTAL_CACHE_DIR="/Users/homans/.cache/crystal"
CRYSTAL_PATH="/usr/local/Cellar/crystal/0.31.0/src:lib"
CRYSTAL_VERSION="0.31.0"
CRYSTAL_LIBRARY_PATH="/usr/local/Cellar/crystal/0.31.0/embedded/lib"
```

amqp-client.cr shard details:
```
  amqp-client:
    github: cloudamqp/amqp-client.cr
    commit: eaabc1558e746b844c6885269edea844a2e373bf
```

**Comment by kalinon** [comment]
Think this is related to https://github.com/crystal-lang/crystal/issues/8216

**Comment by carlhoerberg** [comment]
thanks, looking into it

**Comment by carlhoerberg** [comment]
Yes, it is that bug that you referred too, it will be fixed in 0.31.1, there's a workaround but, it got another bug that I also found: https://github.com/crystal-lang/crystal/issues/8231

**Comment by carlhoerberg** [comment]
Crystal 0.31.1 is out and resolves the issue
